talk/owt/sdk/p2p/p2pclient.cc

```diff
@@ -82,6 +88,45 @@ void P2PClient::RemoveAllowedRemoteId(const std::string& target_id,
   }
   Stop(target_id, on_success, on_failure);
 }
+
+void P2PClient::PublishBatch(const std::string& target_id,
+               std::vector<std::shared_ptr<owt::base::LocalStream>> streams,
+               std::function<void(std::shared_ptr<P2PPublication>, std::string)> on_success,
+               std::function<void(std::unique_ptr<Exception>)> on_failure)
+{
+  // Firstly check whether target_id is in the allowed_remote_ids_ list.
+  bool not_allowed = false;
+  {
+    const std::lock_guard<std::mutex> lock(remote_ids_mutex_);
+    not_allowed = (std::find(allowed_remote_ids_.begin(), allowed_remote_ids_.end(), target_id) ==
+      allowed_remote_ids_.end());
+  }
+  if (not_allowed) {
+    if (on_failure) {
+      std::unique_ptr<Exception> e(
+        new Exception(ExceptionType::kP2PClientRemoteNotAllowed,
+        "Publishing a stream cannot be done since the remote user is not allowed."));
+      on_failure(std::move(e));
+    }
+    return;
+  }
+
+  // Secondly use pcc to publish the streams.
+  auto pcc = GetPeerConnectionChannel(target_id);
+  std::weak_ptr<P2PClient> weak_this = shared_from_this();
+  pcc->PublishBatch(streams, [on_success, weak_this, target_id] (std::shared_ptr<LocalStream> stream) {
```

Comment:
nit: can you update `stream`s signature in the callback to `std::shared_ptr<owt::base::LocalStream>` to match the top-level function signature (line 93)